pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -43,5 +48,17 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 	r.log.Infof("Fixing NSG from %s to %s", oldNSG, correctNSGResourceID)
 	subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
 	err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
-	return err
+	if err != nil {
+		return err
+	}
+
+	return r.updateReconcileSubnetAnnotation(ctx)
```

Comment:
Could we do the update together with the update above, so that we can save one operation?

Reply:
No actually, because the first request is sent to Azure, and the second is to k8s/OpenShift API